lib/WorseReflection/Bridge/TolerantParser/Diagnostics/UnusedImportProvider.php

```diff
@@ -81,8 +81,11 @@ public function exit(NodeContextResolver $resolver, Frame $frame, Node $node): i
         $contents = $node->getFileContents();
 
         foreach ($this->imported as $importedFqn => $imported) {
-            if (isset($this->names[$importedFqn])) {
-                continue;
+            foreach ($this->names as $name => $_) {
+                if (strpos($name, $importedFqn) === 0) {
```

Comment:
what if we had:

```
use Foo as ORMTheSecond;
use Bar as ORM;

/** @ORM\Entity */
```

`ORMTheSecond` would not trigger a diagnostic?

I think I've done this logic somewhere else, I'll check.

Reply:
```
<?php

use Doctrine\Mapping;
use MappingBar;

#[Mapping\Id]
#[MappingBarFoo]
function fo($foo): void
{
}
```

fails, because `MappingBarFoo` starts with `MappingBar`

Reply:
not sure if the best way to do it but:

```
                if (strpos($name . '\\', $importedFqn . '\\') === 0) {
```
seems to fix

Reply:
i think this:

```
            foreach ($this->names as $name => $_) {
                if (strpos($name . '\\', $importedFqn . '\\') === 0) {
                    unset($this->names[$name]);
                    continue 2;
                }
            }
```

although it would be good to avoid the additional loop, maybe we could store the first segment of the names in a property so that we don't need to iterate over all the names again?

Reply:
i tried not adding loop however it doesn't work on docblock =(
```php
<?php

use App\Entity;

/**
 * @param Entity\User $foo
 */
function fo($foo): void
{
}
```